### PR TITLE
double check counts

### PR DESCRIPTION
**Pull request by xithan**
on profile pages and research group pages

### CODE REVIEW

File: mod/profiles/set/ltype_rtype/user/cardtype.rb [hunk 1]
```diff
@@ -7,11 +7,15 @@
 include_set Abstract::WikirateTable
```
Comment (xithan) [inline mod/profiles/set/ltype_rtype/user/cardtype.rb:7]: moved all the stuff below to haml

File: mod/profiles/template/ltype_rtype/user/cardtype/contribution_report.haml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            -if action
+              = report_link report_tab(action), action, true
+            -else
+              %p
+        %li.contribution-report-toggle.text-center.nav-item
+          = report_link toggle_icon, toggle_action, true
+    - if current_tab&.to_sym == :badges
+      = nest badges_earned_card, view: :content
+    - else
+      = contribution_list
```
Comment (xithan) [inline mod/profiles/template/ltype_rtype/user/cardtype/contribution_report.haml]: this replaces more than 10 ruby methods

File: mod/profiles/set/ltype_rtype/user/cardtype.rb [hunk 3]
```diff
@@ -72,13 +77,18 @@ def valid_actions
   end
 
   def has_badges?
-    card.cardtype_card.codename.to_sym.in? Abstract::BadgeSquad::BADGE_TYPES
+    cardtype_codename.in? Abstract::BadgeSquad::BADGE_TYPES
+  end
+
+  def report_title_link
+    link_text = contribution_report_title + nest(badges_earned_card, view: :count)
```
Comment (ethn) [inline mod/profiles/set/ltype_rtype/user/cardtype.rb]: should this be just `report_title`?

File: mod/profiles/set/ltype_rtype/user/cardtype.rb [hunk 4]
```diff
@@ -120,11 +124,14 @@ def toggle_action
   end
 
   def contribution_list
-    return "" unless current_tab && (rcard = report_card(current_tab))
-    nest rcard, view: contribution_list_view, structure: rcard.variant, skip_perms: true
+    if current_tab? :badges
+      nest badges_earned_card, view: :content
+    elsif (rcard = report_card(current_tab))
+      nest rcard, view: contribution_list_view, structure: rcard.variant, skip_perms: true
```
Comment (ethn) [inline mod/profiles/set/ltype_rtype/user/cardtype.rb]: maybe move this do a `contribution_report` method? (just for readability)

File: mod/profiles/template/ltype_rtype/user/cardtype/contribution_report.haml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+        - valid_actions.each do |action|
+          %li.contribution-report-box.nav-item{class: [current_tab?(action) && "active"]}
+            -if action
+              = report_link report_tab(action), action, true
+            -else
+              %p
+        %li.contribution-report-toggle.text-center.nav-item
+          = report_link toggle_ico, toggle_action, true
+    = contribution_list
```
Comment (ethn) [inline mod/profiles/template/ltype_rtype/user/cardtype/contribution_report.haml]: well, this is drastically more readable!!

**Review by ethn — APPROVED** [review]
I may(?) have spotted a typo bug.  Otherwise looks great.

**Pull request merged**